packages/client/src/v2-events/features/events/AdvancedSearch/AdvancedSearch.tsx

```diff
@@ -39,8 +39,9 @@ const messages = defineMessages(messagesToDefine)
 export function AdvancedSearch() {
   const intl = useIntl()
   const allEvents = useEventConfigurations()
-  const location = useLocation()
-  const { searchParams = {}, eventType = '' } = location.state || {}
+  const { eventType = '', ...searchParams } = parse(
+    window.location.search
+  ) as Record<string, string>
```

Comment:
https://api.reactrouter.com/v7/functions/react_router.useSearchParams.html

Reply:
`useSearchParams()` is fine for simple cases, but query-string.parse() is more reliable and simpler, it handles arrays, types, and nested params correctly without manual hacks.

example: 
```
const urlQueryParams = 'child.gender=male&event.registeredAt=2024-05-01&event.registeredAt=2025-05-31'
const [routerSearchParams] = useSearchParams()

routerSearchParams.get("event.registeredAt") // '2024-05-01'
routerSearchParams.getAll("event.registeredAt") // ['2024-05-01', '2024-05-31'] 
Object.fromEntries(routerSearchParams.entries()) // { event.registeredAt: '2024-05-01' }

but we needed { event.registeredAt: ['2024-05-01', '2024-05-31'] }

```